tutorials/crashlytics-logs-to-datadog/index.md

```diff
@@ -0,0 +1,265 @@
+
+1.  Assign the `dataflow.worker` role to allow the service account to allow it to run as a Dataflow worker:
+
+        gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+        --member="serviceAccount:${PIPELINE_SERVICE_ACCOUNT_EMAIL}" \
+        --role=roles/dataflow.worker
+        
+### Create the Cloud Storage bucket
+
+Create a Cloud Storage bucket for storing test data and Dataflow staging location:
+
+    gsutil mb -p ${PROJECT_ID} -l ${REGION_ID} "gs://${GCS_BUCKET}"
+
+
+## Build and launch Dataflow pipeline
+
+Test and compile the pipeline code.
+
+    ./gradlew clean build shadowJar
+
+
+> Add `-x test` flag to skip running tests.
+
+The pipeline supports following options:
+
+| parameter | Default value | description |
+| --- | --- | --- |
+| `sourceBigQueryTableId` | - |  A Fully qualified BigQuery TableId in <b>\<projectId\>:\<datasetId\>.\<tableId\>\[:$\<parition_date\>\]</b> |
+| `bigQuerySqlQuery` | - | The BigQuery SQL query results to send to Datadog. |
+| `shardCount` | **10** | The number of parallel processes to send to Datadog. Using a higher number can result in overloading the Datadog API.|
+| `preserveNulls` | `false` | Allow null values from BigQuery source to be serialized.
+| `datadogApiKey` | - | Provide the API key from [Datadog console](https://app.datadoghq.com/account/settings#api) |
+| `datadogEndpoint` | **https://http-intake.logs.datadoghq.com/v1/input** | Refer Datadog [logging endpoints](https://docs.datadoghq.com/logs/log_collection/?tab=host#logging-endpoints)|
+| `datadogSource` | **crashlytics-bigquery** | Refer to the Datadog [log entry structure](https://docs.datadoghq.com/api/latest/logs/#send-logs) <br> you can customize these parameters to suit your needs |
+| `datadogTags` | **user:crashlytics-pipeline** | Refer to the Datadog [log entry structure](https://docs.datadoghq.com/api/latest/logs/#send-logs) <br> you can customize these parameters to suit your needs |
+| `datadogLogHostname` | **crashlytics** | Refer to the Datadog [log entry structure](https://docs.datadoghq.com/api/latest/logs/#send-logs) <br> you can customize these parameters to suit your needs |
+
+Use only one of `sourceBigQueryTableId` or `bigQuerySqlQuery`.
+
+Define the Crashlytics exported BigQuery table name
+
+    export CRASHLYTICS_BIGQUERY_TABLE="<projectId>:<datasetId>.<tableId>"
+
+You can directly launch the pipeline from the shell using following command:
+
+    bq_2_datadog_pipeline \
+    --project="${PROJECT_ID}" \
+    --region="${REGION_ID}" \
+    --runner="DataflowRunner" \
+    --gcpTempLocation="gs://${GCS_BUCKET}/temp" \
+    --stagingLocation="gs://${GCS_BUCKET}/staging" \
+    --tempLocation="gs://${GCS_BUCKET}/bqtemp" \
+    --datadogApiKey="${DATADOG_API_KEY}" \
+    --sourceBigQueryTableId="${CRASHLYTICS_BIGQUERY_TABLE}"
+    
+You can monitor the Dataflow job on Cloud Console. The pipeline DAG looks as follows:
+
+![Pipeline DAG](pipeline_dag.png)
+
+### Create Dataflow flex template
```

Comment:
As the previous step already contains the command to launch the pipeline, this step may look a bit confusing to some audiences. I would suggest marking this part as optional and also spell out the reason why would one choose to create a flex template over running it directly (like for launching from env without JRE?)

Reply:
Thanks @ktchana: Done. in the latest commit.